src/Microsoft.AspNetCore.Razor.Language/Legacy/CodeBlockEditHandler.cs

```diff
@@ -0,0 +1,132 @@
+// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.Globalization;
+
+namespace Microsoft.AspNetCore.Razor.Language.Legacy
+{
+    internal class CodeBlockEditHandler : SpanEditHandler
+    {
+        public CodeBlockEditHandler(Func<string, IEnumerable<ISymbol>> tokenizer) : base(tokenizer)
+        {
+        }
+
+        protected override PartialParseResultInternal CanAcceptChange(Span target, SourceChange change)
+        {
+            if (IsAcceptableDeletion(target, change))
+            {
+                return PartialParseResultInternal.Accepted;
+            }
+
+            if (IsAcceptableReplacement(target, change))
+            {
+                return PartialParseResultInternal.Accepted;
+            }
+
+            if (IsAcceptableInsertion(change))
+            {
+                return PartialParseResultInternal.Accepted;
+            }
```

Comment:
I assume that you're doing this to follow the structure of the existing stuff 😆

Reply:
yup